frontend/src/metabase/core/components/TabLink/TabLink.tsx

```diff
@@ -3,11 +3,18 @@ import { LinkProps } from "metabase/core/components/Link";
 import { getTabId, getTabPanelId, TabContext } from "../Tab";
 import { TabLinkRoot, TabLabel } from "./TabLink.styled";
 
-export interface TabLinkProps<T> extends LinkProps {
+export interface TabLinkProps<T> extends Omit<LinkProps, "to"> {
   value?: T;
+  to?: string;
```

Comment:
`to` is optional for when you want to use the `TabLinkList` and `TabLink` components without link functionality.